File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/ledger/validations/PositiveSplitAmountValidator.java
package com.decrux.pocketr.api.services.ledger.validations;

import com.decrux.pocketr.api.entities.dtos.CreateSplitDto;
import com.decrux.pocketr.api.exceptions.BadRequestException;
import java.util.List;
import org.springframework.stereotype.Component;

@Component
public class PositiveSplitAmountValidator {

    public void validate(List<CreateSplitDto> splits) {
        for (CreateSplitDto split : splits) {
            if (split.getAmountMinor() <= 0) {
                throw new BadRequestException("All split amounts must be greater than 0");
            }
        }
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/services/ledger/validations/PositiveSplitAmountValidator.kt
package com.decrux.pocketr.api.services.ledger.validations

import com.decrux.pocketr.api.entities.dtos.CreateSplitDto
import com.decrux.pocketr.api.exceptions.BadRequestException
import org.springframework.stereotype.Component

@Component
class PositiveSplitAmountValidator {
    fun validate(splits: List<CreateSplitDto>) {
        splits.forEach { split ->
            if (split.amountMinor <= 0) {
                throw BadRequestException("All split amounts must be greater than 0")
            }
        }
    }
}
